packages/web-app-files/src/helpers/share/link.ts

```diff
@@ -16,9 +16,13 @@ interface CreateQuicklink {
 
 export const createQuicklink = async (args: CreateQuicklink): Promise<Share> => {
   const { resource, store, password, $gettext, ability } = args
+
   const params: { [key: string]: unknown } = {
     name: $gettext('Quicklink'),
-    permissions: ability.can('create-all', 'PublicLink') ? '1' : '0',
+    permissions:
```

Comment:
doesn't work like this, sorry.

a) the permissions always have higher priority. If the default role via capabilities is `viewer` and you don't have the permission to create a public link, the role still needs to be `internal`
b) please use the default role from capabilities to find the corresponding public link role constant and then use the role constant to extract the bitmask (= the value for the permissions).

Reply:
hope its right this way, not sure if I understood b) right